app/src/main/java/org/mozilla/focus/home/HomeFragment.java

```diff
@@ -903,27 +909,56 @@ private static void scheduleRefresh(Handler handler) {
     }
 
     private View.OnClickListener wifiVpnSurveyClickListener = v -> {
-        final ViewGroup rootView = (ViewGroup) v.getParent();
-        if (rootView != null) {
-            LayoutInflater.from(getContext()).inflate(R.layout.wifi_vpn_survey, rootView);
-
-            final View view = rootView.findViewById(R.id.wifi_vpn_root);
-            final TextView textContent = view.findViewById(R.id.wifi_vpn_text_content);
-            final Button btnYes = view.findViewById(R.id.wifi_vpn_btn_yes);
-            final Button btnNo = view.findViewById(R.id.wifi_vpn_btn_no);
-
-            textContent.setText(isSurveyWifiFinding ? R.string.exp_survey_wifi : R.string.exp_survey_vpn);
-            view.setOnClickListener(v1 -> rootView.removeView(view));
-            btnYes.setOnClickListener(v2 -> {
-                textContent.setText(getString(R.string.exp_survey_thanks, "\uD83D\uDE00"));
-                btnYes.setVisibility(View.GONE);
-                btnNo.setVisibility(View.GONE);
-            });
-            btnNo.setOnClickListener(v3 -> {
-                textContent.setText(getString(R.string.exp_survey_thanks, "\uD83D\uDE00"));
-                btnYes.setVisibility(View.GONE);
-                btnNo.setVisibility(View.GONE);
-            });
+        synchronized (this) {
```

Comment:
why do we need synchronized here?

Reply:
In case user tap the survey button very quickly, it might inflate a duplicate view. So I add a synchronized block to protect it.